pkg/kubectl/cmd/convert.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/api/latest"
+	"k8s.io/kubernetes/pkg/api/registered"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	convert_long = `Convert config files between different API versions. Both YAML
+and JSON formats are accepted.
+
+The command takes filename, directory, or URL as input, and convert it into format
+of version specified by --output-version flag. If target version is not specified or
+not supported, convert to latest version.
+
+The default output will be printed to stdout in YAML format. One can use -o option
+to change to output destination.
+`
+	convert_example = `// Convert 'pod.yaml' to latest version and print to stdout.
```

Comment:
We're using `#` for comments now.

Make --local true by default, but allow users to set it to false. `kubectl export` will have the opposite default.

We need to clearly distinguish between commands that just convert the file vs. ones that contact the server.

For --local=false:
Convert the live state of the resource specified by 'pod.yaml' to the latest version and print to stdout.